src/Compiler/CMakeLists.txt

```diff
@@ -24,6 +24,14 @@ add_custom_target(ExternalUtil DEPENDS ${FILE_GENERATE_DIR}/ExternalUtil.hpp)
 
 get_property(OMLibs GLOBAL PROPERTY ONNX_MLIR_LIBS)
 
+# Link LLVM libraries necessary to query which target architectures are configured.
+set(ExtraLLVMLibs)
+list(APPEND ExtraLLVMLibs "LLVMMC")
+llvm_expand_pseudo_components(TARGET_LIBS AllTargetsDescs AllTargetsInfos AllTargetsCodeGens AllTargetsAsmParsers)
```

Comment:
Per the llvm cmake files, the function that should be used here is `llvm_map_components_to_libnames` as it is intended for the final user instead of `llvm_expand_pseudo_components`